Add domain registration checklist step

mattwiebe opened the pull request:
A domain registration step was originally planned for the checklist but scrapped. This lands it.

A simple Guided Tour takes the user through how to search for and select a domain. There are 3 steps. The first one can be removed by clicking/tapping into the search field.

Access the tour at `/domains/add/SITE_SLUG?tour=checklistDomainRegister`

<img width="716" alt="image" src="https://user-images.githubusercontent.com/195089/42969308-6755e63a-8b6b-11e8-96a6-1318346e04c8.png">

<img width="696" alt="image" src="https://user-images.githubusercontent.com/195089/42969325-719fb1b6-8b6b-11e8-828f-ecd360d80339.png">

<img width="691" alt="image" src="https://user-images.githubusercontent.com/195089/42969338-8077d1be-8b6b-11e8-97e2-317bbfe3d0b0.png">

mattwiebe commented:
@fditrapani Especially looking for a high-level overview of the Guided Tour here. Anything you'd like to see improved/added/removed?

jancavan commented:
Drive by comment :)

> A domain registration step was originally planned for the checklist but scrapped.

You still get to this tour through the checklist, correct?

The second tooltip tells me to click "Select", but I only see an "Upgrade" button.
![screenshot 2018-07-19 14 57 19](https://user-images.githubusercontent.com/4924246/42972396-1fc43990-8b64-11e8-9157-8c48d3201efd.png)

This may be edge case, but I see the tooltip on the site picker:
![screenshot 2018-07-19 14 55 16](https://user-images.githubusercontent.com/4924246/42972429-44258b0e-8b64-11e8-93b5-97f3e63fb028.png)

fditrapani commented:
Thanks for putting this together @mattwiebe! Looking through your screenshots, here's some feedback:

Step 1: I remember when I was building this that the prompt got really tricky placing it. Right now it covers the suggestions below. At one point I positioned it static at the top right of the screen but that didn't work on mobile which we disabled shortly after anyways. 

I like the copy here but maybe we should also add something about picking a suggestion. "Type a memorable name in the search field or pick one of the suggestions below to get started. When you find a name that you like, go ahead and press *Select/Upgrade* to continue" (I'm also seeing upgrade like @jancavan. It might be a free plan vs paid plan thing). It would be nice if we can ditch the "next" button and automatically move forward once they select a domain.

Step 2: This step becomes redundant if we go with the suggestion above.

Step 3: I like this one a lot! One of the reasons we gave up this task was because we couldn't get the guided tours to work consistently throughout the entire process. Only thing is I'd suggest changing the wording a little. Rather than "Excellent, you're ~done~ on your way." 

Now from a testing perspective. When I start the tour, my first prompt appears off the screen:
![image](https://user-images.githubusercontent.com/6981253/43004272-391787aa-8bfd-11e8-9163-4ca606bd2b98.png)

fditrapani commented:
> You still get to this tour through the checklist, correct?
That's right @jancavan. Here's what I see:

![image](https://user-images.githubusercontent.com/6981253/43004310-5e91ddf0-8bfd-11e8-836e-413b3e73dac8.png)

> This may be edge case, but I see the tooltip on the site picker:
Can you explain how you got there. I imagine it's because you copied and pasted the url from the description above which would then be an edge case for sure.

fditrapani commented on client/layout/guided-tours/tours/checklist-domain-register-tour.js:
```diff
@@ -0,0 +1,86 @@
+						<span className="tours__completed-icon-wrapper">
+							<Gridicon icon="checkmark" className="tours__completed-icon" />
+						</span>
+						{ translate( 'Excellent, you’re done!' ) }
```
Suggested copy change: Excellent, you're ~done~ on your way!

fditrapani commented on client/layout/guided-tours/tours/checklist-domain-register-tour.js:
```diff
@@ -0,0 +1,86 @@
+					<ButtonRow>
+						<Continue target=".search__input" step="search-results" click hidden />
+						<Next step="search-results" />
+						<SiteLink isButton={ false } href="/checklist/:site">
```
Can we detect people picking a domain rather than having this button?

fditrapani commented on client/layout/guided-tours/tours/checklist-domain-register-tour.js:
```diff
@@ -0,0 +1,86 @@
+			) }
+		</Step>
+
+		<Step name="search-results" arrow="bottom-left" placement="right">
```
I am not sure this step is necessary if we combine the copy with the previous step.

jancavan commented:
> Can you explain how you got there. I imagine it's because you copied and pasted the url from the description above which would then be an edge case for sure.

Yes, that's exactly it, so probably nothing to worry about.

fditrapani commented:
Thanks for putting this together @mattwiebe. We're really close, some minor comments this time:

- The next button doesn't offer any value in the first prompt anymore, can we remove it?
- The call to action (Done) for the second prompt doesn't seem to make sense. Can we change it to "Hide"?
- The call to action (Done) for the second prompt is the only centre aligned button for all the prompts. Can we left align it like the others? I believe you can change this by removing the `<ButtonRow>` wrapper. 
- There's an illustration missing on the stats page when the domain task is featured (Complete all the other tasks by making them complete with the checkbox). Luckily we have an illustration ready to go. Can you add this one in: https://cloudup.com/c9KVKDjv1Rl

mattwiebe commented:
> The next button doesn't offer any value in the first prompt anymore, can we remove it?

I toyed with this, but I thought that perhaps people would want to dismiss it on their own, might be in the way on smaller screens? I removed it in the latest though.

> The call to action (Done) for the second prompt doesn't seem to make sense. Can we change it to "Hide"?

Sounds good to me.

> The call to action (Done) for the second prompt is the only centre aligned button for all the prompts. Can we left align it like the others?

Yup.

> There's an illustration missing on the stats page when the domain task is featured

I've added it and checked of the other items but it's not showing in my list? I looked in the block and I don't see it used anywhere, maybe that was stripped out at some point?

mattwiebe commented on client/layout/guided-tours/tours/checklist-domain-register-tour.js:
```diff
@@ -0,0 +1,82 @@
+				<Fragment>
+					<p>
+						{ translate(
+							'Type a memorable name in the search field or pick one of the suggestions to get started. When you find a name that you like, go ahead and press Select/Upgrade to continue.'
```
Changing to "Pick one of the suggestions below or type a memorable name to get started" as per Slack chat.
